### PR TITLE
Implemented naive status interface

### PR DESCRIPTION
**Pull request by xytis**

### CODE REVIEW

File: lib/resty/t10cache/cache/cache.lua [hunk 1]
```diff
@@ -266,6 +270,25 @@ function _M.purge(self, cache_key)
...
+        ngx.say('Total keys: ' .. status.total .. '\n')
+        if status.preview and #status.preview > 0 then
+            ngx.say('-- cache contents preview --\n')
+            for _,k in pairs(status.preview) do
+                ngx.say(k .. '\n')
+            end
+            ngx.say('-- -- --\n')
+        end
+    else
+        ngx.say('Status page for this cache is not supported')
```
Comment (luza) [inline lib/resty/t10cache/cache/cache.lua]: can't be reached due to check on callbacks.status non-emptiness in new()

File: lib/resty/t10cache/cache/cache.lua [hunk 2]
```diff
@@ -266,6 +270,25 @@ function _M.purge(self, cache_key)
...
 
+function _M.status(self, preview_length)
+    preview_length = preview_length or 0
+    if self.callbacks.status then
+        local status = self.callbacks.status(preview_length)
+        ngx.header['Content-Type'] = 'text/plain'
+        ngx.say('Total keys: ' .. status.total .. '\n')
+        if status.preview and #status.preview > 0 then
+            ngx.say('-- cache contents preview --\n')
+            for _,k in pairs(status.preview) do
```
Comment (luza) [inline lib/resty/t10cache/cache/cache.lua]: can be replaced with table.concat()

File: lib/resty/t10cache/memory.lua [hunk 3]
```diff
@@ -31,6 +31,19 @@ function _M.memory(dict_name, proxy_options)
         purge = function (key)
             dict:delete(key)
             return true
+        end,
+        status = function (preview_length)
+            local keys = dict:get_keys()
```
Comment (luza) [inline lib/resty/t10cache/memory.lua]: get_keys() natively supports specifying max returning keys as its first argument

Comment (luza) [inline lib/resty/t10cache/memory.lua]: ah, we couldn't get the total number this way

File: lib/resty/t10cache/cache/cache.lua [hunk 4]
```diff
@@ -48,11 +48,15 @@ function _M.new(callbacks, opts)
...
+    if callbacks.status and type(callbacks.status) ~= 'function' then
+        return nil, 'status callback is not a function'
+    end
 
     self.callbacks = {
         set   = callbacks.set,
         get   = callbacks.get,
-        purge = callbacks.purge
+        purge = callbacks.purge,
+        status = callbacks.status
```
Comment (xytis) [inline lib/resty/t10cache/cache/cache.lua]: Note that I _am_ assigning nil.

**Pull request merged**